arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -478,7 +478,9 @@ def rule_fr56(
     less than -1000, cf. the Annual Accounts Act §§ 31 and 95 a.
     """
     modelXbrl = val.modelXbrl
-    contextIds = {c.id for c in pluginData.getCurrentAndPreviousReportingPeriodContexts(modelXbrl)}
+    contexts = pluginData.getReportingPeriodContexts(modelXbrl)
+    if contexts is not None:
+        contextIds = {c.id for c in contexts.values()}
```

Comment:
It looks like `contextIds` could be undefined where it's accessed later. Initialize it to an empty set if contexts is `None`.